cpp/include/cudf/io/datasource.hpp

```diff
@@ -340,6 +369,9 @@ class arrow_io_source : public datasource {
   }
 
  private:
+  const std::string uri_start_delimiter = "//";
+  const std::string uri_end_delimiter   = "?";
+  std::shared_ptr<arrow::fs::FileSystem> filesystem;
```

Comment:
Why does `filesystem` need to be a data member? It looks like it isn't used anywhere other than in the ctor.

Reply:
Otherwise it goes out of scope and subsequent `read_host` calls result in segmentation faults

Reply:
Ah, I see.